From 10af4e76550dc7f9f8354a3f9073ddbf60b3099b Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 24 May 2021 07:57:49 +0300
Subject: [PATCH] ar1335: Update the machine dts

Add two separate dts files for each mipi-csi interface.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile              |  2 +
 arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dts    | 88 ++++++++++++++++------
 .../boot/dts/compulab/ucm-imx8m-plus_mipi-csi1.dts | 25 ++++++
 .../boot/dts/compulab/ucm-imx8m-plus_mipi-csi2.dts | 25 ++++++
 4 files changed, 115 insertions(+), 25 deletions(-)
 create mode 100644 arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi1.dts
 create mode 100644 arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi2.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 62ee304cb6f3..80d395953adc 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -1,4 +1,6 @@
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus.dtb
+dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus_mipi-csi1.dtb
+dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus_mipi-csi2.dtb
 
 always-y       := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dts
index fb0f6cf72d0a..bd3255bed5e9 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dts
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dts
@@ -557,10 +557,31 @@
 	pinctrl-0 = <&pinctrl_i2c5>;
 	status = "okay";
 
-	eeprom@54 {
-		compatible = "atmel,24c08";
-		reg = <0x54>;
-		pagesize = <16>;
+	ar1335_af_mipi1: ar1335_af_mipi1@42 {
+		compatible = "econ,ar1335_af", "ar1335_af";
+		reg = <0x42>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_csi_mclk>;
+		clocks = <&clk IMX8MP_CLK_IPP_DO_CLKO2>;
+		clock-names = "csi_mclk";
+		assigned-clocks = <&clk IMX8MP_CLK_IPP_DO_CLKO2>;
+		assigned-clock-parents = <&clk IMX8MP_CLK_24M>;
+		assigned-clock-rates = <24000000>;
+		csi_id = <0>;
+		pwn-gpios = <&pca9555 8 GPIO_ACTIVE_HIGH>;
+		rst-gpios = <&pca9555 9 GPIO_ACTIVE_LOW>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		mipi_csi;
+		status = "disabled";
+
+		port {
+			ar1335_af_mipi1_ep: endpoint {
+				data-lanes = <1 2 3 4>;
+				clock-lanes = <0>;
+				remote-endpoint = <&mipi1_sensor_ep>;
+			};
+		};
 	};
 
 	goodix_ts@5d {
@@ -585,29 +606,29 @@
 	pinctrl-0 = <&pinctrl_i2c6>;
 	status = "okay";
 
-	ov5640: ov5640_mipi@3c {
-		compatible = "ovti,ov5640";
-		reg = <0x3c>;
+	ar1335_af_mipi2: ar1335_af_mipi1@42 {
+		compatible = "econ,ar1335_af", "ar1335_af";
+		reg = <0x42>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_csi0_pwn>, <&pinctrl_csi_mclk>;
+		pinctrl-0 = <&pinctrl_csi_mclk>;
 		clocks = <&clk IMX8MP_CLK_IPP_DO_CLKO2>;
-		clock-names = "xclk";
+		clock-names = "csi_mclk";
 		assigned-clocks = <&clk IMX8MP_CLK_IPP_DO_CLKO2>;
 		assigned-clock-parents = <&clk IMX8MP_CLK_24M>;
 		assigned-clock-rates = <24000000>;
-		csi_id = <0>;
-		powerdown-gpios = <&pca9555 8 GPIO_ACTIVE_LOW>;
-		reset-gpios = <&pca9555 3 GPIO_ACTIVE_HIGH>;
+		csi_id = <0x01>;
+		pwn-gpios = <&pca9555 12 GPIO_ACTIVE_HIGH>;
+		rst-gpios = <&pca9555 13 GPIO_ACTIVE_LOW>;
 		mclk = <24000000>;
 		mclk_source = <0>;
 		mipi_csi;
-		status = "okay";
+		status = "disabled";
 
 		port {
-			ov5640_mipi_0_ep: endpoint {
-				remote-endpoint = <&mipi_csi0_ep>;
-				data-lanes = <1 2>;
+			ar1335_af_mipi2_ep: endpoint {
+				data-lanes = <1 2 3 4>;
 				clock-lanes = <0>;
+				remote-endpoint = <&mipi2_sensor_ep>;
 			};
 		};
 	};
@@ -1294,38 +1315,55 @@
 &mipi_csi_0 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-	status = "okay";
+	status = "disabled";
 
 	port@0 {
 		reg = <0>;
-		mipi_csi0_ep: endpoint {
-			remote-endpoint = <&ov5640_mipi_0_ep>;
-			data-lanes = <2>;
-			csis-hs-settle = <13>;
+		mipi1_sensor_ep: endpoint@1 {
+			remote-endpoint = <&ar1335_af_mipi1_ep>;
+			data-lanes = <4>;
+			csis-hs-settle = <23>;
 			csis-clk-settle = <2>;
 			csis-wclk;
 		};
 	};
 };
 
+&mipi_csi_1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "disabled";
+
+	port@1 {
+		reg = <1>;
+		mipi2_sensor_ep: endpoint {
+			remote-endpoint = <&ar1335_af_mipi2_ep>;
+			data-lanes = <4>;
+			csis-hs-settle = <23>;
+			csis-clk-settle = <2>;
+			csis-wclk;
+		};
+	};
+};
+
+
 &cameradev {
 	status = "okay";
 };
 
 &isi_0 {
-	status = "okay";
+	status = "disabled";
+	fsl,chain_buf;
 
 	cap_device {
 		status = "okay";
 	};
 
-	m2m_device {
-		status = "okay";
-	};
 };
 
 &isi_1 {
 	status = "disabled";
+	fsl,chain_buf;
 
 	cap_device {
 		status = "okay";
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi1.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi1.dts
new file mode 100644
index 000000000000..f600a0b7c494
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi1.dts
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 CompuLab
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8M-Plus with mipi csi1 enabled";
+	compatible = "compulab,ucm-imx8m-plus", "fsl,imx8mp";
+};
+
+&ar1335_af_mipi1 {
+	status = "okay";
+};
+
+&mipi_csi_0 {
+	status = "okay";
+};
+
+&isi_0 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi2.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi2.dts
new file mode 100644
index 000000000000..5ae73e2f185e
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus_mipi-csi2.dts
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 CompuLab
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8M-Plus with mipi csi2 enabled";
+	compatible = "compulab,ucm-imx8m-plus", "fsl,imx8mp";
+};
+
+&ar1335_af_mipi2 {
+	status = "okay";
+};
+
+&mipi_csi_1 {
+	status = "okay";
+};
+
+&isi_1 {
+	status = "okay";
+};
-- 
2.11.0

